sns_driver.go

```diff
@@ -0,0 +1,93 @@
+package es
+
+import (
+	"encoding/json"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sns"
+)
+
+// NewSNSDriver creates an SNSDriver
+func NewSNSDriver(client *sns.SNS, topicArn string, driver Driver) *SNSDriver {
+	return &SNSDriver{
+		client:   client,
+		topicArn: topicArn,
+		driver:   driver,
+	}
+}
+
+// SNSDriver creates a driver decorator that sends a notification to AWS' SNS
+// when new events are saved.
+type SNSDriver struct {
+	client   *sns.SNS
+	topicArn string
+	driver   Driver
+}
+
+type snsPacket struct {
+	Types []string
+}
```

Comment:
Thinking of adding EventID as pointer here?

Reply:
To better match existing terminology, rename this struct to snsMessage, or snsMessageBody if conflicting?

Reply:
> Thinking of adding EventID as pointer here?

I thought about it… but then, as we're grouping the event types under a single collection, having the IDs as a separate collection felt out of place. Besides, I'm not sure if having the IDs would help in any way. Having said that, perhaps adding the _last_ event ID can be helpful for the consumers to know whether they should ignore the notification.

> To better match existing terminology, rename this struct to snsMessage, or snsMessageBody if conflicting?

Yeah… `snsMessage` sounds good.